spec/unit/middleware/timeout_spec.rb

```diff
@@ -0,0 +1,177 @@
+require 'spec_helper'
+require 'qless/job'
+require 'qless/middleware/timeout'
+
+module Qless
+  module Middleware
+    ::Rspec.describe Timeout do
+      class JobClass
+        attr_accessor :perform
+
+        def around_perform_call_counter
+          @around_perform_call_counter ||= 0
+        end
+
+        def around_perform(job)
+          @around_perform_call_counter = (@around_perform_call_counter || 0) + 1
+        end
+      end
+
+      let(:kernel_class) { class_double(Kernel) }
+
+      let(:job) {
+        instance_double(Qless::Job)
+      }
+
+      def make_worker(timeout_class, timeout_seconds, kernel_class)
+        Class.new(JobClass) do
+          include Qless::Middleware::Timeout.new(timeout_class: timeout_class,
+              kernel_class: kernel_class) { timeout_seconds }
+        end.new
+      end
+
+      before {
+        allow(job).to receive(:reconnect_to_redis)
+        allow(job).to receive(:klass_name)
+        allow(job).to receive(:fail)
+        allow(kernel_class).to receive(:exit!)
```

Comment:
I think the specs here could benefit from using `instance_double(...).as_null_object` or `instance_spy(...)` (the latter is only available on recent versions of RSpec 3.  Then you wouldn't need to explicitly stub these methods, and in the specs below you could stick with the familiar arrange-act-assert pattern by using `expect(...).to have_received(...)` rather than ahead-of-time `expect(...).to receive(...)`.

If you want to make that change it would almost certainly help to upgrade to RSpec 3 as it's support for using doubles as spies is much better.